### PR TITLE
feature: support kingbase

### PR DESCRIPTION
**Pull request by yougecn**
人大金仓 kingbase适配

- [ ] I have registered the PR [changes](../changes).

### CODE REVIEW

File: rm-datasource/src/main/java/org/apache/seata/rm/datasource/exec/kingbase/KingbaseInsertExecutor.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    /**
+     * get primary key values by default
+     * @return the pk values
+     * @throws SQLException the sql exception
+     */
+    @Override
+    @Deprecated
+    public List<Object> getPkValuesByDefault() throws SQLException {
+        // current version 1.2 only support postgresql.
```
Comment (funky-eyes) [inline rm-datasource/src/main/java/org/apache/seata/rm/datasource/exec/kingbase/KingbaseInsertExecutor.java]:
```suggestion
```

File: rm-datasource/src/main/java/org/apache/seata/rm/datasource/exec/kingbase/KingbaseInsertExecutor.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    /**
+     * get primary key values by default
+     *
+     * @param pkKey the pk key
+     * @return the primary values
+     * @throws SQLException the sql exception
+     */
+    @Override
+    public List<Object> getPkValuesByDefault(String pkKey) throws SQLException {
+        // current version 1.2 only support postgresql.
```
Comment (funky-eyes) [inline rm-datasource/src/main/java/org/apache/seata/rm/datasource/exec/kingbase/KingbaseInsertExecutor.java]:
```suggestion
```

File: rm-datasource/src/main/java/org/apache/seata/rm/datasource/sql/handler/kingbase/KingbaseEscapeHandler.java [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+import org.apache.seata.sqlparser.util.JdbcConstants;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The type kingbase keyword checker.
+ *
+ * @author yougecn
```
Comment (funky-eyes) [inline rm-datasource/src/main/java/org/apache/seata/rm/datasource/sql/handler/kingbase/KingbaseEscapeHandler.java]:
```suggestion
```

File: rm-datasource/src/main/java/org/apache/seata/rm/datasource/exec/kingbase/KingbaseInsertExecutor.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The type Postgresql insert executor.
+ *
+ * @author  yougecn
```
Comment (funky-eyes) [inline rm-datasource/src/main/java/org/apache/seata/rm/datasource/exec/kingbase/KingbaseInsertExecutor.java]:
```suggestion
 * @author  yougecn
```
所有@author都要去除
All @authors must be removed

**Review by funky-eyes — COMMENTED** [review]
测试用例要添加一些
Test cases to add some

File: rm-datasource/src/test/java/org/apache/seata/rm/datasource/sql/druid/kingbase/KingbaseSelectForUpdateRecognizerTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+public class KingbaseSelectForUpdateRecognizerTest {
+
+    private static final String DB_TYPE = "oracle";
```
Comment (funky-eyes) [inline rm-datasource/src/test/java/org/apache/seata/rm/datasource/sql/druid/kingbase/KingbaseSelectForUpdateRecognizerTest.java]: DB_TYPE为什么是oracle?
Why DB_TYPE oracle?

File: rm-datasource/src/test/java/org/apache/seata/rm/datasource/sql/druid/kingbase/KingbaseUpdateRecognizerTest.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+public class KingbaseUpdateRecognizerTest {
+
+    private static final String DB_TYPE = "oracle";
```
Comment (funky-eyes) [inline rm-datasource/src/test/java/org/apache/seata/rm/datasource/sql/druid/kingbase/KingbaseUpdateRecognizerTest.java]: ditto

Comment (yougecn) [reply rm-datasource/src/test/java/org/apache/seata/rm/datasource/sql/druid/kingbase/KingbaseUpdateRecognizerTest.java]: ok 了

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**